config.go

```diff
@@ -137,10 +136,6 @@ func (c Config) BinaryRepositoryURL(binaryRepositoryURL string) Config {
 	return c
 }
 
-func (c Config) GetConnectionURL() string {
```

Comment:
We won't be able to remove this without bumping a major version, which I think we're unlikely to want to do for this change. Can you try and work it into here?

Reply:
I already pitched a solution attempt within config.go and we both agreed that there should be no major logic there.